app/src/main/java/com/asfoundation/wallet/promotions/voucher/DiamondsButtonsAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package com.asfoundation.wallet.promotions.voucher
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AbsListView
+import android.widget.BaseAdapter
+import android.widget.Button
+import android.widget.GridView
+import com.asf.wallet.R
+
+
+class DiamondsButtonsAdapter(
+    val context: Context,
+    val buttonModels: List<DiamondsButtonModel>,
+    val onclick: OnClick
+) :
+    BaseAdapter() {
```

Comment:
Normally this should be RecyclerView.Adapter<ViewHolder> instead and you would basically implement onCreateViewHolder and onBindViewHolder. You can check PromotionsAdapter for an example. This also makes it so you don't need getItem(), nor getItemId(), nor getView().